### DISCUSSION TITLE
Backwards compatibility

### DISCUSSION THREAD

**Discussion by adw96** [statdivlab/rigr#65] [General]
For veteran uses of uwIntroStats (and for us to let folx teaching classes using the package know in advance), what can we expect in terms of backwards compatibility? Could we please list functions here for which the input or output has changed significantly? @cwolock @taylorokonek

**Comment by cwolock** [comment]
`ttest`: This function no longer performs tests of proportions - it only performs t-tests using a data vector as input. The proportion tests (exact and asymptotic) using a data vector (i.e., a vector of 0's and 1's) are being branched off into a new function called `proptest` (currently being implemented). 

`ttesti`: Similar to above. Performing a test of proportions on summary data (i.e., total number of successes and number of trials, as opposed to a data vector of 1's and 0's) will be branched into `proptesti`. 

I don't gather many people were using these functions for proportion tests, but this is the biggest change I've made so far in terms of backwards compatibility. 

Also, the entire group of testing functions (`wilcoxon`, `ttest`, `ttesti`, `proptest`, `proptesti`) will now return an object instead of only printing the results. You can still run 

`ttest(data)`

and the formatted results will be printed. But you could also run 

`my_test <- ttest(data)`

You can then `print(my_test)`, or directly access the results using the usual `$` syntax. This was mainly implemented in order to facilitate unit tests, but if you need those objects for any reason, they'll now be accessible to you.

**Comment by taylorokonek** [comment]
`regress`: Will no longer support gee or proportional hazards regression. A number of parameters in the function are thus no longer relevant and are in the process of being removed (`id`, `ties`, and `strata`, for example). 

One change in output is that p-values and test statistics are now clearly labeled for logistic regression and Poisson regression. The default value for the parameter `useFdstn` is TRUE (and always has been), which means test statistics, p-values, and robust confidence intervals are all calculated using a t distribution rather than a normal distribution, even for logistic and Poisson regression. The output labeling now reflects this, whereas before it would label p-values as "Pr(>|z|)" regardless of whether the t distribution or normal distribution was used.

No changes made to `descrip` should affect backwards compatibility at this point.

**Reply by taylorokonek** [reply]
In `regress`, the user must now specify the full `fnctl` argument rather than relying on string matching. For example, they must write out `fnctl = 'geometric mean'` rather than simply `fnctl = 'geom'` or `fnctl = 'g'`.

**Reply by taylorokonek** [reply]
All `lspline` capabilities will be removed, and cannot be included in the formula for a call to `regress`. Both `dummy` and `polynomial` will still be maintained.

**Reply by taylorokonek** [reply]
also in `regress`, the parameters `model.x`, `model.y`, and `model.f` are no longer available to change. The default values of `FALSE`, `FALSE`, and `TRUE` are now used throughout.
